IdentityCore/tests/MSIDAccountCredentialsCacheTests.m

```diff
@@ -2755,6 +2755,41 @@ - (void)testRemoveCredentialsWithQuery_whenQueryIsNotExactMatch_andAccessTokensQ
     XCTAssertTrue([remainignItems count] == 0);
 }
 
+- (void)testRemoveCredentialsWithQuery_whenQueryIsNotExactMatch_andAccessTokensQuery_shouldRemoveAllExpiredItems
+{
+    MSIDCredentialCacheItem *token1 = [self createTestAccessTokenCacheItem];
+    token1.homeAccountId = @"uid.utid1";
+    token1.expiresOn = [NSDate date];
+    [self saveItem:token1];
+
+    MSIDCredentialCacheItem *token2 = [self createTestAccessTokenCacheItem];
+    token2.homeAccountId = @"uid.utid1";
+    token2.realm = @"contoso2.com";
+    token2.expiresOn = [[NSDate date] dateByAddingTimeInterval:60];
+    [self saveItem:token2];
+
+    [self saveItem:[self createTestRefreshTokenCacheItem]];
+    
+    [NSThread sleepForTimeInterval:5.0];
```

Comment:
Instead of sleeping for 5 seconds can the token.expiresOn be set 5 seconds earlier from now : [NSDate dateWithTimeIntervalSinceNow:-5.0]?